src/commands.ts

```diff
@@ -140,22 +142,95 @@ export function registerCommands(context: vscode.ExtensionContext, prManager: Pu
+	}));
+
 	context.subscriptions.push(vscode.commands.registerCommand('pr.openOriginalFile', async (e: GitFileChangeNode) => {
 		// if this is an image, encode it as a base64 data URI
 		const imageDataURI = await asImageDataURI(e.parentFilePath, prManager.repository);
 		vscode.commands.executeCommand('vscode.open', imageDataURI || e.parentFilePath);
 	}));
 
-	context.subscriptions.push(vscode.commands.registerCommand('pr.openModifiedFile', (e: GitFileChangeNode) => {
-		vscode.commands.executeCommand('vscode.open', e.filePath);
-	}));
+	/**
+	 * Retrieves all GitFileChangeNodes under given node.
+	 * @param parentNode Files category or directory node
+	 */
+	async function getGitFileChangeNodes(parentNode: FilesCategoryNode | DirectoryTreeNode): Promise<GitFileChangeNode[]> {
+		// Helper function to recursively populate provided array with GitFileChangeNodes
+		function populateGitFileChangeNodesHelper(nodeArr: GitFileChangeNode[], node: FilesCategoryNode | DirectoryTreeNode | GitFileChangeNode) {
+			if (node instanceof GitFileChangeNode) {
+				nodeArr.push(node);
+			} else if (node instanceof FilesCategoryNode) {
+				node.directories.forEach(child => {
+					if (child instanceof DirectoryTreeNode ||
+						child instanceof GitFileChangeNode) {
+						populateGitFileChangeNodesHelper(nodeArr, child);
+					}
+				});
+			} else if (node instanceof DirectoryTreeNode) {
+				node.children.forEach(child => {
+					if (child instanceof DirectoryTreeNode ||
+						child instanceof GitFileChangeNode) {
+						populateGitFileChangeNodesHelper(nodeArr, child);
+					}
+				});
+			}
+		}
 
-	context.subscriptions.push(vscode.commands.registerCommand('pr.openDiffView', async (fileChangeNode: GitFileChangeNode | InMemFileChangeNode) => {
+		const fileChangeNodes: GitFileChangeNode[] = [];
+		populateGitFileChangeNodesHelper(fileChangeNodes, parentNode);
+
+		return fileChangeNodes;
+	}
+
+	/**
+	 * Executes provided command for all GitFileChangeNodes under the given node.
+	 * @param e Parent node
+	 * @param command Function to be executed on for each GitFileChangeNodes
+	 */
+	async function executeAll(e: DirectoryTreeNode | FilesCategoryNode, command: (node: GitFileChangeNode) => Promise<void>) {
+		// Minimum number of files for warning message to be shown
+		const MIN_FILES_FOR_WARNING = 10;
+		const messageOptions: vscode.MessageOptions = {
+			modal: true
+		};
+
+		const nodes: GitFileChangeNode[] = await getGitFileChangeNodes(e);
+
+		// Ask user for confirmation when opening large number of files
+		if (nodes.length >= MIN_FILES_FOR_WARNING) {
+			vscode.window.showWarningMessage(`Are you sure you want to open ${nodes.length} files?`, messageOptions, 'Yes').then(async option => {
+				if (option === 'Yes') {
+					// Ensure files are processed in the same order
+					for (const node of nodes) {
+						await command(node);
+					}
+				}
+			});
+		} else {
+			// Ensure files are processed in the same order
```

Comment:
I'm torn on this - I like having the files opened in order, but as you said, it does make the action take a noticeable amount of time. I think I prefer the snappiness of opening them in parallel to having them in exactly tree order. @JacksonKearl @rebornix thoughts?

Reply:
It's a good idea but opening tons of editors and switching between them automatically looks a bit scary to me.

Having them opening in order sounds right to me as I would love to see them being opened in the predictable order. However, only the first editor should be opened and then focused, all following editors should only be hidden tabs. You can select multiple files in the File Explorer and right click to open them, you can see that they are all opened but you are only focusing in the first one.

I'm not sure yet if our API allows that but IMHO that's the right behavior if we want to take this feature in.

Reply:
I agree, it should function like selecting files in Explorer and clicking **Open to the Side**.

Found some relevent code [here](https://github.com/microsoft/vscode/blob/87064db1033a9d8d768cd7ed676083413c8a222c/src/vs/workbench/contrib/files/browser/fileCommands.ts#L286) where that command is registered.

I tried calling `vscode.commands.executeCommand('explorer.openToSide', ___);` but it expects a single resource URI as the argument, not a list of URIs. It gets the list of URIs to open from the File, List, and Editor Services. If I call `explorer.openToSide` multiple times, a new column is created in the editor window every time, which is not what we want.

Does some additional work need to be done in `vscode` then?